### PR TITLE
Updates -- Modern babel/webpack/etc.

### PR DESCRIPTION
**Pull request by ryan-roemer**
* Depends on https://github.com/FormidableLabs/builder-react-component-dev/pull/1
* Depends on https://github.com/FormidableLabs/builder-react-component/pull/43
* Upgrade Travis to modern yarn + node engines.
* Remove `.jsx` suffix.
* Update a whole lot of dependencies
* Output new es/ directory
* Add publishr support.

/cc @kevinmstephens @aweary @rgerstenberger

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,21 +13,35 @@
...
     "version": "builder run npm:version",
+    "postversion": "builder run npm:postversion",
+    "postpublish": "builder run npm:postpublish",
     "test": "builder run npm:test"
   },
   "dependencies": {
-    "builder": "^3.0.0",
-    "builder-react-component": "^0.3.5"
+    "builder": "^3.2.2",
+    "builder-react-component": "FormidableLabs/builder-react-component#chore/updates"
```
Comment (ryan-roemer) [inline package.json]: WIP: Depends on upstream PRs.

File: package.json [hunk 2]
```diff
@@ -13,21 +13,35 @@
...
   },
   "devDependencies": {
-    "builder-react-component-dev": "^0.3.5",
-    "enzyme": "^2.4.1",
-    "lodash": "^4.13.1",
-    "react": "^15.2.1 || ^0.14.8",
-    "react-addons-test-utils": "^15.2.1 || ^0.14.8",
-    "react-dom": "^15.2.1 || ^0.14.8"
+    "builder-react-component-dev": "FormidableLabs/builder-react-component-dev#chore/updates",
```
Comment (ryan-roemer) [inline package.json]: WIP: Depends on upstream PRs.

File: demo/list-item.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+export default class ListItem extends React.Component {
```
Comment (kevinmstephens) [inline demo/list-item.js]: This could be a stateless functional component but of course this is perfectly good as well.

**Review by kevinmstephens — COMMENTED** [review]
I got an error on `npm i` using node 6 and npm 3. 

```
[builder:proc:start] Command: babel src --copy-files -d lib
SyntaxError: src/components/typeahead.js: Unexpected token (14:21)
  12 |
  13 | export default class RadonTypeahead extends React.Component {
> 14 |   static displayName = "RadonTypeahead";
     |                      ^
  15 |
  16 |   static propTypes = {
  17 |     inputComponent: PropTypes.element,
[builder:proc:end:1] Command: babel src --copy-files -d lib
```

File: demo/list-item.js (hunk 3 above)
Comment (ryan-roemer) [reply demo/list-item.js]: Sure. I'll take a stab at that.

**Comment by ryan-roemer** [comment]
@kevinmstephens -- Haha, the `npm i` thing is because I had unpushed changes. It works now.

**Review by kevinmstephens — APPROVED** [review]
`npm i` working and smoke test LGTM.

**Comment by ryan-roemer** [comment]
@kevinmstephens -- Changed `ListItem`. Ready for re-review. Also, somehow Travis wasn't turned on for this repo, so closing and re-opening this PR to kick travis

**Comment by kevinmstephens** [comment]
List item change LGTM as well.

**Comment by ryan-roemer** [comment]
Released in `radon-typeahead@0.5.0`

**Pull request merged**